include/private/vkd3d_string.h

```diff
@@ -37,4 +37,14 @@ char *vkd3d_strdup_n(const char *str, size_t n);
 WCHAR *vkd3d_wstrdup(const WCHAR *str);
 WCHAR *vkd3d_wstrdup_n(const WCHAR *str, size_t n);
 
+static inline bool vkd3d_string_ends_with_n(const char *str, size_t str_len, const char *ending, size_t ending_len)
+{
+    return str_len >= ending_len && !strcmp(str + (str_len - ending_len), ending);
```

Comment:
_n naming here is somewhat misleading, since I would expect that to only look at len number of characters (i.e. strncmp).